src/3d/framegraph/qgsframegraph.h

```diff
@@ -239,34 +202,43 @@ class QgsFrameGraph : public Qt3DCore::QEntity
     */
     void removeClipPlanes();
 
+    /**
+     * Registers a new the render view \a renderView with name \a name
+     *
+     * Will take ownership of the renderView
+     */
+    bool registerRenderView( std::shared_ptr<QgsAbstractRenderView> renderView, const QString &name );
```

Comment:
I am confused now - why is shared pointer being passed here, while the docstring says the framegraph takes ownership of the render view? :thinking: 

Generally I don't like shared pointers unless really needed, because it is hard to track where are all the places where the shared pointer can be kept, and when it finally gets deleted... It seems it would be simpler if you passed just raw pointer here, and keep the statement that the render view ownership will be taken...

Reply:
The main idea with the shared_ptr is to make the code safer for example to avoid any accidental deletion when someone retrieve a renderview.

Reply:
What I also wanted to say is that `shared_ptr` is very rarely used in QGIS APIs, and I do not see a particular reason to force shared pointers in this case. The ownership handling should be clear here - if some code calls `registerRenderView()`, then it should pass ownership to the frame graph. The same pattern happens in many places in QGIS APIs (pretty much all registries - e.g. `QgsSymbolLayerRegistry` has `addSymbolLayerType()` and `removeSymbolLayerType()` - one takes ownership, and the other one destroys the instance)

Reply:
ok after internal discussion I will switch to your solution